### PR TITLE
Extract action creators and selectors so they can be tested.

### LINKED ISSUE #563: test new `mapStateToProps` and `mapDispatchToProps` functions from session management work as expected
i grouped these together as one ticket, since they're all small and similar.

- [x] test `mapStateToProps` and `mapDispatchToProps` in `LoginPanel`
- [x] test `Editor`'s `mapStateToProps`
- [x] unit test for `ImportResourceTemplate.mapStateToProps`

spawned from #528

### PR DESCRIPTION
**Pull request by jcoyne**
Fixes #563

### CODE REVIEW

File: src/selectors.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+
```
Comment (ndushay) [inline src/selectors.js]: "selectors" seems an odd name choice, esp as it's overloaded per CSS selectors.  Maybe these could just be in CognitoUtils?

File: __tests__/actionCreators.test.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import { authenticationFailed, authenticationSucceeded, signedOut } from '../src/actionCreators';
+
+describe('authenticationFailed', () => {
+  const currentUser = { hello: 'world' }
+  const errInfoauthenticate = { foo: 'bar' }
+
+  const authResult = {
+    currentUser: currentUser,
+    authenticationError: errInfoauthenticate
+  };
```
Comment (ndushay) [inline __tests__/actionCreators.test.js]: OMG - the dreaded semicolon!   (slightly prefer without them; not a biggie)

File: src/selectors.js (hunk 1 above)
Comment (jcoyne) [reply src/selectors.js]: Selectors is a term of art in Redux: https://medium.com/@matthew.holman/what-is-a-redux-selector-a517acee1fe8

File: __tests__/actionCreators.test.js (hunk 2 above)
Comment (jcoyne) [reply __tests__/actionCreators.test.js]: Should we add that to es-lint?

Comment (jcoyne) [reply __tests__/actionCreators.test.js]: I added a change to eslint for this: #590

**Comment by jcoyne** [comment]
@jermnelson I've renamed that file as you suggested.

**Pull request merged**